mysql-test/suite/rocksdb/t/ttl_primary_read_filtering.test

```diff
@@ -0,0 +1,245 @@
+--source include/have_rocksdb.inc
+--source include/have_debug.inc
+
+--disable_warnings
+DROP TABLE IF EXISTS t1;
+--enable_warnings
+
+SET @@global.rocksdb_update_cf_options = 'default={disable_auto_compactions=true;};';
+
+# The purpose of read filtering for tables with TTL is to ensure that during a
+# transaction a key which has expired already but not removed by compaction
+# yet, is not returned to the user.
+#
+# Without this the user might be hit with problems such as disappearing rows
+# within a transaction, etc, because the compaction filter ignores snapshots
+# when filtering keys.
+
+# Basic read filtering test
+CREATE TABLE t1 (
+  a int PRIMARY KEY
+) ENGINE=rocksdb
+COMMENT='ttl_duration=1;';
+
+INSERT INTO t1 values (1);
+INSERT INTO t1 values (2);
+set global rocksdb_force_flush_memtable_now=1;
```

Comment:
There should be some extra test cases. Instead of flushing, you can leave the keys in the memtable perhaps to guarantee they are not removed through the compaction filter.

Then you can do the following: set ttl_duration to 1 second. sleep 2 seconds. select * should return nothing. Disable filtering. select * should return everything. enable filtering, select * should return nothing. This explicitly tests the filtering code and you know the rows exist.

Alternatively, you can do the following:

transaction 1, create a snapshot and select * => returns nothing.
transaction 2, insert into table, flush
transaction 1, select * => returns nothing, but the snapshot should prevent the compaction code from removing the rows, no matter what the ttl duration is.
transaction 2, select * -> sees nothing, disable filter, select * -> sees everything, enable filter, select * -> sees nothing.